src/app/shared/services/broker.service.ts

```diff
@@ -74,23 +74,49 @@ export class BrokerService {
     };
   }
 
+  importProjectUsingGeo(accession: string): Observable<any> {
+    const params = {
+      'accession': accession,
+    };
+    return this.http
+      .post(`${this.API_URL}/import-geo-project`, null, {params})
+      .pipe(
+        catchError((errorResponse: HttpErrorResponse) => {
+          return throwError(errorResponse.error);
+        })
+      )
+  }
+
   downloadSpreadsheetUsingGeo(geoAccession: string): Observable<any> {
     const params = {
       'accession': geoAccession,
     };
     return this.http
       .post(`${this.API_URL}/import-geo`, null,
-        {params, responseType: 'blob', observe: 'response'}).pipe(
+        {params, responseType: 'blob', observe: 'response'})
+      .pipe(
+        catchError(this.parseErrorBlob),
         map(response => {
-          if (response.status == HttpStatusCode.Ok) {
-            return this.getFileDataFromResponse(response)
-          } else {
-            throw throwError(response);
+          if(response.status == HttpStatusCode.Ok){
+            return this.getFileDataFromResponse(response);
           }
         })
       );
   }
 
+  parseErrorBlob(err: HttpErrorResponse): Observable<any> {
+    const reader: FileReader = new FileReader();
+
+    const obs = new Observable((observer: any) => {
+      reader.onloadend = (e) => {
+        observer.error(JSON.parse(reader.result as string));
```

Comment:
I think it's better to use `throwError`

Reply:
why?

Reply:
i think it's more readable yeah, applied.